Send deregistration event from AMF to UDM

bmeglicit opened the pull request:
[UDM,UDR] Add support for endpoint for patching subscription data
[AMF] Send deregistration event to UDM

When UE sends deregistration request, AMF needs to send a
Nudm_UECM_Deregistration request to UDM.
The order of requests is now the following:
- send PDU session release to SMF
- send AM policy control release to PCF
- send deregistration event to UDM

acetcom commented:
Hi @bmeglicit 

I haven't been able to properly analyze this pull request because I've been busy with other things. So, please review the following points first.

There are many errors below.
```
06/13 23:08:15.948: [core] ERROR: OpenAPI_patch_item_convertToJSON() failed [value] (../lib/sbi/custom/patch_item.c:80)
```

And the TS23.502 standard says to operate in the following order.

1. send PDU session release to SMF
2. send deregistration event to UDM
3. send AM policy control release to PCF

Could you please check it one more time?

Thanks a lot!
Sukchan

bmeglicit commented:
@acetcom Will check it out and fix it. Thanks for checking!

bmeglicit commented:
Changes:
- Fixed a minor bug when converting PatchItem to JSON (did not see an error amongst the debug log messages before)
- Changed order of requests - deregistration event to UDM is sent before AM policy control release to PCF
- Added additional check to UDM - Guami in the ModificationRequest must match to the stored Guami
